tests/ui/cloned_last.rs

```diff
@@ -0,0 +1,36 @@
+// run-rustfix
+#![warn(clippy::all)]
+
+fn main() {
+    let _: Option<String> = vec!["1".to_string(), "2".to_string(), "3".to_string()]
```

Comment:
How about `["1", "2", "3"].into_iter().cloned().collect::<Vec<_>>()` to avoid the `to_string()`s?

Also perhaps put it in a binding and clone as needed.

Reply:
```rust
143 |                     let x = ["1", "2", "3"].into_iter().cloned().collect::<Vec<_>>();
    |                                                         ^^^^^^ the trait `Clone` is not implemented for `str`
```

Reply:
I could write it like this:
```
let _ = ["1", "2", "3"].into_iter().map(ToString::to_string).collect::<Vec<_>>()
```

Reply:
Although, that would deny the purpose of having this test.